kasa/smart/smartdevice.py

```diff
@@ -787,3 +790,91 @@ def presets(self) -> list[BulbPreset]:
     def has_effects(self) -> bool:
         """Return True if the device supports effects."""
         return "LightEffectModule" in self.modules
+
+    # Plug / Wall Switch methods
+
+    @property
+    def has_led(self) -> bool:
+        """Return True if device has a led."""
+        return "LedModule" in self.modules
+
+    @property
+    def led(self) -> bool:
+        """Return the state of the led."""
+        if not self.has_led:
+            raise KasaException("Device does not support led.")
+        return cast(LedModule, self.modules["LedModule"]).led
+
+    async def set_led(self, state: bool):
+        """Set the state of the led (night mode)."""
+        if not self.has_led:
+            raise KasaException("Device does not support led.")
```

Comment:
Controlling the led on/off is so unimportant, that we should get rid of it in the common API. We could deprecate it, but I'd rather just break the API for 0.7 release and instruct library users to use the feature interface for this.

Reply:
Now moved to common `LedModule`.  Will not actually break the existing interface until a consumer tries to move from `SmartPlug` > `Device` and they can then do:
```py
if module := device.get_module(kasa.modules.LedModule):
	await module.set_led(True)
```